### PR TITLE
CDN Module rewrite

### PR DESCRIPTION
**Pull request by thomast1906**
### Change description ###
Redesign of CDN profile(s) to create on a team basis rather than one CDN profile per environment. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: environments/sbox/sbox.tfvars [hunk 1]
```diff
@@ -18,6 +18,7 @@ cft_apps_cluster_ips   = ["10.10.1.250", "10.10.3.250"]
 
 frontends = [
   {
+    team             = "idam"
```
Comment (timja) [inline environments/sbox/sbox.tfvars:21]: we normally use the term `product` to refer to a group of applications

File: modules/shutter_page/cdn_endpoints.tf [hunk 2]
```diff
@@ -1,18 +1,21 @@
 resource "azurerm_cdn_endpoint" "shutter_endpoint" {
-  count                  = length(var.shutter_apps)
-  name                   = "hmcts-${split(".", replace(var.shutter_apps[count.index].custom_domain, "www.", ""))[0]}-shutter-${var.env}"
-  profile_name           = var.cdn_profile
+    for_each = { for frontend in var.shutter_apps : frontend.name => frontend
```
Comment (timja) [inline modules/shutter_page/cdn_endpoints.tf:2]: nice 🎉

File: modules/shutter_page/storage_container.tf [hunk 3]
```diff
@@ -1,6 +0,0 @@
-resource "azurerm_storage_container" "containers" {
```
Comment (timja) [inline modules/shutter_page/storage_container.tf:1]: how is this happening now?

File: components/shutter/cdn_profile.tf [hunk 4]
```diff
@@ -1,8 +1,8 @@
-resource "azurerm_cdn_profile" "main" {
-  name                = "hmcts-shutter-${var.env}"
-  location            = "West US"
-  resource_group_name = var.shutter_rg
-  sku                 = var.cdn_sku
+# resource "azurerm_cdn_profile" "main" {
```
Comment (timja) [inline components/shutter/cdn_profile.tf:1]: should this PR be draft? I assume this isn't ready yet?

File: modules/shutter_page/storage_container.tf (hunk 3 above)
Comment (naikajah) [reply modules/shutter_page/storage_container.tf:1]: This is happening by default now when you enable the static website

File: components/shutter/cdn_profile.tf (hunk 4 above)
Comment (naikajah) [reply components/shutter/cdn_profile.tf:1]: This should be removed entirely instead of commented out.

File: azure_pipeline.yaml [hunk 5]
```diff
@@ -183,15 +183,15 @@ parameters:
...
-    - deployment: 'prod_shutter' 
-      environment: 'prod'
-      component: 'shutter'
-      service_connection: 'dcd-cftapps-prod'
-      storage_account_rg: 'core-infra-prod-rg'
-      storage_account_name: 'cftappsprod'
-      dependsOn: 'sbox_cftapps_cluster_lb'
-      files_list: 'components/shutter modules/shutter_page environments/prod/prod.tfvars environments/variables.tf'
-      alwaysApply: false
+    # - deployment: 'prod_shutter'
```
Comment (timja) [inline azure_pipeline.yaml:186]: I assume this is commented out till you manage to migrate the existing ones?

**Review by naikajah — APPROVED** [review]

**Pull request merged**